gui/src/main.rs

```diff
@@ -286,13 +325,31 @@ impl App for Gui {
                 });
             }
         };
+        if self.config.import_window_open == true {
+            egui::Window::new("Import Modal")
+                .open(&mut self.config.import_window_open)
+                .show(ctx, |ui| {
+                    ui.label("Please copy and paste the file path to import");
+                    ui.horizontal(|ui| {
+                        ui.text_edit_singleline(&mut self.config.import_file_path);
+                        if ui.button("Import").clicked() {
+                            self.rt.block_on(async {
+                                let path = self.config.import_file_path.to_owned();
+                                let _ = match self.config.import_mode {
+                                    ImportMode::COLLECTION => {
+                                        PostieApi::import_collection(&path).await
+                                    }
+                                    ImportMode::ENVIRONMENT => todo!(),
+                                };
+                            });
+                        };
+                    });
+                });
+        }
     }
 }
 
-#[tokio::main]
```

Comment:
Had to remove this since it was causing network requests and parsing to crash the app, due to trying to spawn a blocking thread within a blocking thread.
```
Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.
```

I'm pretty sure that annotating a function with #[tokio::main] essentially does turns the function into this:

```
fn main() {
    tokio::runtime::Builder::new_multi_thread()
        .enable_all()
        .build()
        .unwrap()
        .block_on(async {
            ...
        })
}
```

but the Gui struct spawns a blocking thread already,
```
impl Default for Gui {
    fn default() -> Self {
        Self {
            config: GuiConfig::default(),
            rt: runtime::Builder::new_multi_thread()
                .enable_all()
                .build()
                .unwrap(),
        }
    }
}
```


im not 100% sure of the best way to manage this situation, whether its trying to use tokio::spawn to leverage other threads? for the mean time I think we'll just need to keep all db stuff on the back end side?